lib/types/protocol.go

```diff
@@ -12,7 +12,7 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-package common
+package types
 
 type Protocol struct {
```

Comment:
Do they belong in the "net" package then, along with `IP` and `IPNet`?

Or, should there be specific "api" and "backend" implementations of all of these?  If we're sharing structs between the `api` and `backend` packages aren't we breaking the isolation to an extent? Sort of plays into what we want to do about the IP and IPNet structs we've defined (I think Kubernetes treats these as strings on the API, so maybe we should do the same and just convert to back as needed?)